apps/mark-scan/accessible-controller/src/controllerd.rs

```diff
@@ -394,17 +393,23 @@ fn main() {
                     exit(0);
                 }
                 match port.read(serial_buf.as_mut_slice()) {
-                    Ok(size) => {
-                        if let Err(e) = handle_command(&mut device, &serial_buf[..size]) {
-                            log!(
-                                event_id: EventId::UnknownError,
-                                message: format!(
-                                    "Unexpected error when handling controller command: {e}"
-                                ),
-                                event_type: EventType::SystemStatus
-                            );
+                    Ok(size) => match handle_command(&serial_buf[..size]) {
+                        Ok(key_option) => {
+                            if key_option.is_some() {
```

Comment:
Refactored `handle_command` to avoid passing the `device`, which is cumbersome to create in CI because it requires special permissions. This seemed easier than configuring udev rules for CI.

The downside is I introduced this nested conditioning to handle when the command should no-op ie. when the controller indicates one of its buttons was released. We only want to send keyboard events when the button is depressed.

To support this, `handle_command` returns `Some(keyboard::Key)` on controller button press, and `None(keyboard::Key)` on button release.

 Open to feedback on how to handle these checks more cleanly/inline.

Reply:
This can be improved by being more specific about the types you're looking for in your match arms:

```rust
                        Ok(Some(key)) => {
                            if let Err(err) = send_key(&mut device, key) {
                                log!(EventId::UnknownError, "Error sending key: {err}");
                            }
                        }
                        Ok(None) => {}
```

In general, you don't want to use `is_some()` or `is_ok()` et al when you want the value being wrapped. Use `match`/`if let`/`let/else` destructuring